### DISCUSSION TITLE
Problem about compiling the application

### DISCUSSION THREAD

**Discussion by K-Komal-98** [idaholab/moose#22047] [Q&A Getting Started] [answered]
Earlier, my application got compiled then I updated the conda and moose according to this link "https://mooseframework.inl.gov/getting_started/installation/update_moose.html". Now, I am getting error. I repeated the process but the error continues to exist. Kindly help me with this issue. I have attached the screenshots of the error. 
![Screenshot from 2022-09-08 10-37-06](https://user-images.githubusercontent.com/113084320/189039025-1e871900-a390-4ee0-a698-ac8657d78aef.png)
![Screenshot from 2022-09-08 10-36-03](https://user-images.githubusercontent.com/113084320/189039031-c53fe80d-ee9a-4397-a30d-4393d53040b4.png)

**Comment by GiudGiud** [comment, marked as answer]
Hello

See this post
https://github.com/idaholab/moose/discussions/21936

You need to update the packages using mamba. 
More information on updating (everything, but you dont need to update moose necessarily) there:
https://mooseframework.inl.gov/getting_started/installation/update_moose.html
Guillaume

**Reply by K-Komal-98** [reply]
I updated the packages using mamba but the error is still there. I followed exactly the same instructions given in the link.

**Reply by milljm** [reply]
I've seen this on Ubuntu (22.04)... Can you provide a `mamba list`? And also the version of Ubuntu you are using.

Thanks!

**Reply by milljm** [reply]
Also, if you can refrain from posting pictures of text to these discussions it will help others down the road when performing 'searches' of possibly the same issue you are experiencing.

**Reply by K-Komal-98** [reply]
I am working on Ubuntu (22.04). I reinstalled MOOSE and now I am not getting this error but a different one. The error is as follows:

Linking Executable /home/komal/projects/nucleation_bm/nucleation_bm-opt...
/home/komal/mambaforge3/envs/moose/bin/../lib/gcc/x86_64-conda-linux-gnu/10.4.0/../../../../x86_64-conda-linux-gnu/bin/ld: /home/komal/projects/nucleation_bm/lib/libnucleation_bm-opt.so: undefined reference to `InputParameters validParams<Material>()'
collect2: error: ld returned 1 exit status
make: *** [/home/komal/projects/moose/framework/app.mk:416: /home/komal/projects/nucleation_bm/nucleation_bm-opt] Error 1

**Reply by K-Komal-98** [reply]
Can you please help me with this error?

**Reply by ats5662** [reply]
This worked for me 
```
mamba activate moose
mamba update --all
sudo apt-get install lsb-core
cd ~/projects/moose
unset LIBMESH_DIR
./scripts/update_and_rebuild_libmesh.sh
```

**Reply by GiudGiud** [reply]
this is a conda-based install, you **should not** run the scripts

this error on validParams is likely due to the deprecation of the validParams template. See this post
https://mooseframework.inl.gov/newsletter/2020_04.html#!

is your app (nuclearion_bm?) open source? So I could try to build it locally

Guillaume

**Comment by K-Komal-98** [comment]
I reinstalled many times and now finally the problem is solved. Thankyou!
